codalab/worker_manager/gcp_batch_worker_manager.py

```diff
@@ -0,0 +1,157 @@
+        )
+
+    def __init__(self, args):
+        super().__init__(args)
+
+        self.codalab_username = os.environ.get('CODALAB_USERNAME')
+        self.codalab_password = os.environ.get('CODALAB_PASSWORD')
+        if not self.codalab_username or not self.codalab_password:
+            raise EnvironmentError(
+                'Valid credentials need to be set as environment variables: CODALAB_USERNAME and CODALAB_PASSWORD'
+            )
+
+        # Authenticate via GCP
+        credentials: service_account.Credentials = service_account.Credentials.from_service_account_file(
+            self.args.credentials_path, scopes=['https://www.googleapis.com/auth/cloud-platform']
+        )
+
+        cluster_manager_client: ClusterManagerClient = ClusterManagerClient(credentials=credentials)
+        cluster = cluster_manager_client.get_cluster(
+            name=f'projects/{self.args.project}/locations/{self.args.zone}/clusters/{self.args.cluster}'
+        )
+
+        # Save SSL certificate to connect to the GKE cluster securely
+        cert_path = os.path.join(self.args.cert_path, 'gke.crt')
+        with open(cert_path, 'wb') as f:
+            f.write(base64.b64decode(cluster.master_auth.cluster_ca_certificate))
+
+        # Configure and initialize Kubernetes client
+        configuration: client.Configuration = client.Configuration()
+        configuration.host = f'https://{cluster.endpoint}:443'
+        configuration.api_key = {'authorization': f'Bearer {credentials.token}'}
+        configuration.verify_ssl = True
+        configuration.ssl_ca_cert = cert_path
+        client.Configuration.set_default(configuration)
+
+        self.k8_client: client.ApiClient = client.ApiClient(configuration)
+        self.k8_api: client.CoreV1Api = client.CoreV1Api(self.k8_client)
+
+    def get_worker_jobs(self) -> List[WorkerJob]:
+        try:
+            # Fetch the running pods
+            pods: client.V1PodList = self.k8_api.list_namespaced_pod(
+                'default', field_selector='status.phase==Running'
+            )
+            logger.debug(pods.items)
+            return [WorkerJob(True) for _ in pods.items]
+        except client.ApiException as e:
+            logger.error(f'Exception when calling Kubernetes CoreV1Api->list_namespaced_pod: {e}')
+            return []
+
+    def start_worker_job(self) -> None:
+        # This needs to be a unique directory since jobs may share a host
+        work_dir_prefix: str = (
+            self.args.worker_work_dir_prefix if self.args.worker_work_dir_prefix else '/tmp/'
+        )
+        worker_id: str = uuid.uuid4().hex
+        worker_name: str = f'cl-worker-{worker_id}'
+        work_dir: str = os.path.join(work_dir_prefix, f'{worker_name}_work_dir')
+        command: List[str] = self.build_command(worker_id, work_dir)
+        worker_image: str = 'codalab/worker:' + os.environ.get('CODALAB_VERSION', 'latest')
+
+        config: Dict[str, Any] = {
```

Comment:
a bit late but it might be a good idea to use a template from a yaml file, execute it, and maybe pass that into kubernetes. 

I'm not sure what the python best practice is (maybe jinja) but I would think something like this: https://golang.org/pkg/text/template/